homeassistant/components/progettihwsw/binary_sensor.py

```diff
@@ -0,0 +1,95 @@
+"""Control binary sensor instances."""
+
+from datetime import timedelta
+import logging
+
+from ProgettiHWSW.input import Input
+import async_timeout
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import setup_input
+from .const import DEFAULT_POLLING_INTERVAL_SEC, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set the progettihwsw platform up and create sensor instances (legacy)."""
+
+    return True
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the binary sensors from a config entry."""
+    board_api = hass.data[DOMAIN][config_entry.entry_id]
+    input_count = config_entry.data["input_count"]
+    binary_sensors = []
+
+    async def async_update_data():
+        """Fetch data from API endpoint of board."""
+        async with async_timeout.timeout(5):
+            return await board_api.get_inputs()
+
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
```

Comment:
this should include the domain of your integration

Reply:
Will do.